### ISSUE TITLE
Error building libp2p with dcutr feature

### ISSUE THREAD

**Issue by haider-rs** [libp2p/rust-libp2p#2833]
## Summary
Cannot build libp2p with dcutr feature.
Created a new cargo project
Added a dependency of libp2p 
`libp2p = { version = "0.46.1", features = ["tcp-tokio", "dns-tokio"] }`
run cargo check, runs fine, change the dependency 
`libp2p = { version = "0.46.1", features = ["tcp-tokio", "dns-tokio", "dcutr"] }`
throws error

## Expected behaviour
should return Finished dev

## Actual behaviour
returns 
```
error: failed to select a version for `signature`.
    ... required by package `ecdsa v0.13.3`
    ... which satisfies dependency `ecdsa-core = "^0.13"` of package `p256 v0.10.0`
    ... which satisfies dependency `p256 = "^0.10.0"` of package `libp2p-core v0.34.0`
    ... which satisfies dependency `libp2p-core = "^0.34.0"` (locked to 0.34.0) of package `libp2p v0.46.1`
    ... which satisfies dependency `libp2p = "^0.46.1"` (locked to 0.46.1) of package `testing_env v0.1.0 (/Users/mp-haidera-pyse-403/Desktop/testing_env)`
versions that meet the requirements `>=1.3.1, <1.5` are: 1.4.0, 1.3.2, 1.3.1

all possible versions conflict with previously selected packages.

  previously selected package `signature v1.6.0`
    ... which satisfies dependency `signature = ">=1.3.1"` (locked to 1.6.0) of package `ed25519 v1.5.2`
    ... which satisfies dependency `ed25519 = "^1"` (locked to 1.5.2) of package `ed25519-dalek v1.0.1`
    ... which satisfies dependency `ed25519-dalek = "^1.0.1"` (locked to 1.0.1) of package `libp2p-core v0.34.0`
    ... which satisfies dependency `libp2p-core = "^0.34.0"` (locked to 0.34.0) of package `libp2p v0.46.1`
    ... which satisfies dependency `libp2p = "^0.46.1"` (locked to 0.46.1) of package `testing_env v0.1.0 (/Users/mp-haidera-pyse-403/Desktop/testing_env)`

failed to select a version for `signature` which could resolve this conflict
```

<details><summary>Debug Output</summary>
<p>

```
<output>
```
</p>
</details>

## Version

- libp2p version (version number, commit, or branch):
0.44.0
0.46.0
0.46.1

**Comment by elenaf9** [comment]
Could you try running `cargo update` and see if the error still occurs?

**Comment by haider-rs** [comment]
thank you, the error is gone.

**Comment by haider-rs** [comment]
@elenaf9 though that fixed the error in a new project. I am still getting error even on running cargo update in my project and it still throws the same error:

```
error: failed to select a version for `signature`.
    ... required by package `ecdsa v0.14.4`
    ... which satisfies dependency `ecdsa-core = "^0.14"` of package `k256 v0.11.0`
    ... which satisfies dependency `k256 = "^0.11"` of package `ethers-core v0.17.0`
    ... which satisfies dependency `ethers-core = "^0.17.0"` of package `ethers v0.17.0`
    ... which satisfies dependency `ethers = "^0.17.0"` of package `tesseract-connector v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract/connector)`
    ... which satisfies path dependency `connector` (locked to 0.0.1) of package `tesseract-node v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract)`
versions that meet the requirements `>=1.5, <1.7` are: 1.6.0, 1.5.0

all possible versions conflict with previously selected packages.

  previously selected package `signature v1.3.1`
    ... which satisfies dependency `signature = ">=1.3.1, <1.5"` of package `ecdsa v0.13.3`
    ... which satisfies dependency `ecdsa-core = "^0.13"` of package `p256 v0.10.0`
    ... which satisfies dependency `p256 = "^0.10.0"` of package `libp2p-core v0.34.0`
    ... which satisfies dependency `libp2p-core = "^0.34.0"` of package `libp2p v0.46.1`
    ... which satisfies dependency `libp2p = "^0.46.1"` of package `tesseract-network v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract/network)`
    ... which satisfies path dependency `network` (locked to 0.0.1) of package `tesseract-node v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract)`

failed to select a version for `signature` which could resolve this conflict
```
any leads why? cant even run cargo update

**Comment by elenaf9** [comment]
> cant even run cargo update

What is the output when you run `cargo update`?

**Comment by haider-rs** [comment]
> > cant even run cargo update
> 
> What is the output when you run `cargo update`?

it throws the same error

```
Updating crates.io index
error: failed to select a version for `signature`.
    ... required by package `ecdsa v0.14.4`
    ... which satisfies dependency `ecdsa-core = "^0.14"` of package `k256 v0.11.0`
    ... which satisfies dependency `k256 = "^0.11"` of package `ethers-core v0.17.0`
    ... which satisfies dependency `ethers-core = "^0.17.0"` of package `ethers v0.17.0`
    ... which satisfies dependency `ethers = "^0.17.0"` of package `tesseract-connector v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract/connector)`
    ... which satisfies path dependency `connector` of package `tesseract-node v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract)`
versions that meet the requirements `>=1.5, <1.7` are: 1.6.0, 1.5.0

all possible versions conflict with previously selected packages.

  previously selected package `signature v1.3.1`
    ... which satisfies dependency `signature = ">=1.3.1, <1.5"` of package `ecdsa v0.13.3`
    ... which satisfies dependency `ecdsa-core = "^0.13"` of package `p256 v0.10.0`
    ... which satisfies dependency `p256 = "^0.10.0"` of package `libp2p-core v0.34.0`
    ... which satisfies dependency `libp2p-core = "^0.34.0"` of package `libp2p v0.46.1`
    ... which satisfies dependency `libp2p = "^0.46.1"` of package `tesseract-network v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract/network)`
    ... which satisfies path dependency `network` of package `tesseract-node v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract)`

failed to select a version for `signature` which could resolve this conflict
```

**Comment by elenaf9** [comment]
I don't think this is a libp2p error. Looks more like a cargo issue; potentially related to this?: https://github.com/rust-lang/cargo/issues/9240. Could you try to manually delete your `Cargo.lock` and run it again?

**Comment by haider-rs** [comment]
> I don't think this is a libp2p error. Looks more like a cargo issue; potentially related to this?: [rust-lang/cargo#9240](https://github.com/rust-lang/cargo/issues/9240). Could you try to manually delete your `Cargo.lock` and run it again?

yeah I also think its a cargo issue since cargo update solved it in new project.
but I have tried deleting cargo.lock, target, even deleted ~/.cargo/registry didn't help yet.

**Comment by haider-rs** [comment]
> it throws the same error
> 
> ```
> Updating crates.io index
> error: failed to select a version for `signature`.
>     ... required by package `ecdsa v0.14.4`
>     ... which satisfies dependency `ecdsa-core = "^0.14"` of package `k256 v0.11.0`
>     ... which satisfies dependency `k256 = "^0.11"` of package `ethers-core v0.17.0`
>     ... which satisfies dependency `ethers-core = "^0.17.0"` of package `ethers v0.17.0`
>     ... which satisfies dependency `ethers = "^0.17.0"` of package `tesseract-connector v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract/connector)`
>     ... which satisfies path dependency `connector` of package `tesseract-node v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract)`
> versions that meet the requirements `>=1.5, <1.7` are: 1.6.0, 1.5.0
> 
> all possible versions conflict with previously selected packages.
> 
>   previously selected package `signature v1.3.1`
>     ... which satisfies dependency `signature = ">=1.3.1, <1.5"` of package `ecdsa v0.13.3`
>     ... which satisfies dependency `ecdsa-core = "^0.13"` of package `p256 v0.10.0`
>     ... which satisfies dependency `p256 = "^0.10.0"` of package `libp2p-core v0.34.0`
>     ... which satisfies dependency `libp2p-core = "^0.34.0"` of package `libp2p v0.46.1`
>     ... which satisfies dependency `libp2p = "^0.46.1"` of package `tesseract-network v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract/network)`
>     ... which satisfies path dependency `network` of package `tesseract-node v0.0.1 (/Users/mp-haidera-pyse-403/Desktop/tesseract)`
> 
> failed to select a version for `signature` which could resolve this conflict
> ```

if you see this error it says ethers indirectly requires signature version >=1.5, <1.7
but libp2p indirectly requires signature version >=1.3.1, <1.5 so we have a version conflict here.

**Comment by elenaf9** [comment]
Ah seems like the reason is that our `p256` dependency is outdated, which we only use on feature `ecdsa`. 
Pending PR to fix that is #2636, though it seems like this is currently blocked. 
Since we don't really need the `ecdsa` feature in `libp2p-dcutr` a workaround to fix this now is to disable the default features for the `libp2p-core` dependency (like we already do in every other package that depends on `libp2p-core`). Opened #2836.

**Comment by elenaf9** [comment]
@Haider-Ali-DS could you test if #2836 fixes the error for you?

**Comment by haider-rs** [comment]
> @Haider-Ali-DS could you test if #2836 fixes the error for you?

yup it did. Thank you

**Comment by mxinden** [comment]
@Haider-Ali-DS @elenaf9 thanks for the help on this. Not disabling default features on the import was my mistake. I released `libp2p-dcutr` `v0.5.1` with @elenaf9 's fix.

**Comment by mxinden** [comment]
For those that need `ecdsa` support, I just released `libp2p-core` `v0.35.1` with `p256` `v0.11.0`.

With these two releases, I think this is fully solved. Please comment in case I am missing something.

Again, thanks for the team work here.
